### PR TITLE
MM-19238 Migrate tests from 'model/team_test.go' to use testify

### PR DESCRIPTION
**Pull request by pikami**
#### Summary
Migrate tests from 'model/team_test.go' to use testify

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/12632

### CODE REVIEW

File: model/team_test.go [hunk 1]
```diff
@@ -6,63 +6,56 @@ package model
...
+	require.NotNil(t, err, "should be invalid")
 
 	o.Name = "zzzzz"
 	o.Type = TEAM_OPEN
 	o.InviteId = NewId()
-	if err := o.IsValid(); err != nil {
-		t.Fatal(err)
-	}
+	err = o.IsValid()
+	require.Nil(t, err, err)
```
Comment (levb) [inline model/team_test.go]: nit: why display the error twice?

Comment (ali-farooq0) [reply model/team_test.go]: should this be `require.Nil(t, err, err.Error())` instead?

File: model/team_test.go [hunk 2]
```diff
@@ -95,9 +88,8 @@ var domains = []struct {
 
 func TestValidTeamName(t *testing.T) {
 	for _, v := range domains {
-		if IsValidTeamName(v.value) != v.expected {
-			t.Errorf("expect %v as %v", v.value, v.expected)
-		}
+		actual := IsValidTeamName(v.value)
+		assert.Equal(t, actual, v.expected)
```
Comment (ali-farooq0) [inline model/team_test.go]: i believe the `Equal` api expects `expected` first and then actual. https://godoc.org/github.com/stretchr/testify/assert#Equal. So this should be flipped

File: model/team_test.go [hunk 3]
```diff
@@ -112,24 +104,20 @@ var tReservedDomains = []struct {
 
 func TestReservedTeamName(t *testing.T) {
 	for _, v := range tReservedDomains {
-		if IsReservedTeamName(v.value) != v.expected {
-			t.Errorf("expect %v as %v", v.value, v.expected)
-		}
+		actual := IsReservedTeamName(v.value)
+		assert.Equal(t, actual, v.expected)
```
Comment (ali-farooq0) [inline model/team_test.go]: same comment as above, needs to be flipped.
https://godoc.org/github.com/stretchr/testify/assert#Equal

File: model/team_test.go [hunk 4]
```diff
@@ -112,24 +104,20 @@ var tReservedDomains = []struct {
...
-		t.Fatal("didn't clean name properly")
-	}
+	actual = CleanTeamName("Admin Really cool")
+	require.Equal(t, actual, "really-cool", "didn't clean name properly")
 
-	if CleanTeamName("super-duper-guys") != "super-duper-guys" {
-		t.Fatal("didn't clean name properly")
-	}
+	actual = CleanTeamName("super-duper-guys")
+	require.Equal(t, actual, "super-duper-guys", "didn't clean name properly")
```
Comment (ali-farooq0) [inline model/team_test.go]: all of these should be updated to follow `require.Equal(t, expected, actual, ...)`

File: model/team_test.go (hunk 1 above)
Comment (pikami) [reply model/team_test.go]: the previous logic used `err` as a message so I used the same logic

File: model/team_test.go (hunk 2 above)
Comment (pikami) [reply model/team_test.go]: fixed.

File: model/team_test.go (hunk 3 above)
Comment (pikami) [reply model/team_test.go]: fixed.

File: model/team_test.go (hunk 4 above)
Comment (pikami) [reply model/team_test.go]: fixed.

**Comment by pikami** [comment]
> 
> 
> Thanks for your contribution @pikami 🎉 ! The require.Equal needs to be updated to follow `require.Equal(t, expected, actual)` as per their api https://godoc.org/github.com/stretchr/testify/require#Equal

Thanks, Fixed

**Comment by levb** [comment]
@pikami Can you please sync & merge the master branch (`git pull <mattermost-remote> master`)? There also appears to be a (transient?) build failure - maybe it gets resolved with the sync?

**Comment by pikami** [comment]
> 
> 
> @pikami Can you please sync & merge the master branch (`git pull <mattermost-remote> master`)? There also appears to be a (transient?) build failure - maybe it gets resolved with the sync?

synced

**Pull request merged**